### PR TITLE
Enable less tests for pull requests

### PR DESCRIPTION
**Pull request by holke**
**_Describe your changes here:_**

The `--enable-less-tests` configure option is activated for pull requests.
For the nightly builds of main, the option should still be disabled.

**_All these boxes must be checked by the reviewers before merging the pull request:_**

As a reviewer please read through all the code lines and make sure that the code is fully understood, bug free, well-documented and well-structured.

- [x] The author added a BSD statement to `doc/` (or already has one)
- [x] The code compiles without warning in debugging and release mode, with and without MPI (this should be executed automatically in a github action)

  If the Pull request introduces code that is not covered by the github action (for example coupling with a new library):
  - [ ] Should this use case be added to the github action?
  - [ ] If not, does the specific use case compile and all tests pass (check manually)

- [x] All tests pass (in various configurations, this should be executed automatically in a github action)
- [ ] New source/header files are properly added to the Makefiles
- [x] The reviewer executed the new code features at least once and checked the results manually
- [ ] The code is covered in an existing or new test case
- [ ] New tests use the Google Test framework
- [x] The code follows the [t8code coding guidelines](https://github.com/holke/t8code/wiki/Coding-Guideline)
- [x] The code is well documented
- [x] All function declarations, structs/classes and their members have a proper doxygen documentation
- [x] All new algorithms and data structures are sufficiently optimal in terms of memory and runtime (If this should be merged, but there is still potential for optimization, create a new issue)
- [ ] Testing of this template: If you feel something is missing from this list, contact the developers

### CODE REVIEW

**Comment by holke** [comment]
Please check carefully: 

 - Does `t8code tests / build (pull_request)` have the less-test option enabled?
 - Does `t8code tests / build (push)` not have the option enabled?
 
 If we merge this, we need to check the nightly builds of main if they still have the long test suite enabled.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -264,8 +264,11 @@ jobs:
     #       we get errors in configure that we cannot remove (for example the autotools way
     #       of checking for libm results in a compiler warning).
     #       Thus, we add '-Werror' later at the make step.
+    - name: less-test-option
+      if: github.event_name == 'pull_request'
+      run: export LESS_TEST_OPTION="--enable-less-tests"
     - name: build config variables
-      run: export CONFIG_OPTIONS="--without-blas"
+      run: export CONFIG_OPTIONS="--without-blas $LESS_TEST_OPTION"
```
Comment (holke) [inline .github/workflows/tests.yml:271]:
```suggestion
      run: export CONFIG_OPTIONS="--without-blas ${LESS_TEST_OPTION}"
```

**Comment by holke** [comment]
This did not work. Am i using the `if` in the workflow file correctly?

**Comment by Davknapp** [comment]
> Please check carefully:
> 
> * Does `t8code tests / build (pull_request)` have the less-test option enabled?
> * Does `t8code tests / build (push)` not have the option enabled?
> 
> If we merge this, we need to check the nightly builds of main if they still have the long test suite enabled.

I triggered the workflow manually and the less-tests-option was not set.

**Review by Davknapp — APPROVED** [review]

**Comment by Davknapp** [comment]
If you don't want to add anything else, this is ready to merge

**Pull request merged**